apisix/plugins/tencent-cloud-cls/cls-sdk.lua

```diff
@@ -62,8 +62,15 @@ local params_cache = {
 local function get_ip(hostname)
     local _, resolved = socket.dns.toip(hostname)
     local ip_list = {}
-    for _, v in ipairs(resolved.ip) do
-        insert_tab(ip_list, v)
+    if not resolved.ip then
```

Comment:
How about use https://github.com/apache/apisix/blob/master/apisix/core/resolver.lua#L61 to resolve domain?

Reply:
The original code used `socket.dns.topip`, I didn't modify it.  What is the difference between these two methods?

Reply:
You could check logic between the two functions, we always use `parse_domain`, maybe it will resolve the problem of the `socket.dns.toip` occasionally returns nil

Reply:
This pr is ok, but we need to know the reason

Reply:
@jiangfucheng PTAL

Reply:
@Sn0rt Ok, give me some time to test it.

Reply:
@jiangfucheng do you have any update for this ?

Reply:
Sorry, there is currently no conclusion yet.